packages/core/src/entity/EntityStoreBase.ts

```diff
@@ -27,23 +30,150 @@ export class EntityStoreBase {
     this.addEntity(entity);
   }
 
+  public getEntryOrAsset(
+    linkOrEntryOrAsset: UnresolvedLink<'Entry' | 'Asset'> | Asset | Entry,
+    // link: UnresolvedLink<'Entry' | 'Asset'>,
+    path: string,
+  ): Entry | Asset | undefined {
+    const resolveFieldset = (
+      unresolvedFieldset: Array<[null, string, string?]>,
+      headEntry: Entry | Asset,
+    ) => {
+      const resolvedFieldset: Array<[Entry | Asset, string, string?]> = [];
+      let entityToResolveFieldsFrom: Entry | Asset = headEntry;
+      for (let i = 0; i < unresolvedFieldset.length; i++) {
+        const isLeaf = i === unresolvedFieldset.length - 1; // with last row, we are not expecting a link, but a value
+        const row = unresolvedFieldset[i];
+        const [, field, _localeQualifier] = row;
+        if (!entityToResolveFieldsFrom) {
+          throw new Error(
+            `Logic Error: Cannot resolve field ${field} of a fieldset as there is no entity to resolve it from.`,
+          );
+        }
+        if (isLeaf) {
+          resolvedFieldset.push([entityToResolveFieldsFrom, field, _localeQualifier]);
+          break;
+        }
+
+        const fieldValue = get<string | UnresolvedLink<'Entry' | 'Asset'>>(
+          entityToResolveFieldsFrom,
+          ['fields', field],
+        );
+
+        if (undefined === fieldValue) {
+          return {
+            resolvedFieldset,
+            isFullyResolved: false,
+            reason: `Cannot resolve field Link<${entityToResolveFieldsFrom.sys.type}>(sys.id=${entityToResolveFieldsFrom.sys.id}).fields[${field}] as field value is not defined`,
+          };
+        } else if (isLink(fieldValue)) {
+          const entity = this.getEntityFromLink(fieldValue);
+          if (entity === undefined) {
+            return {
```

Comment:
I switched this from throwing an error to returning a reason, as in editor mode, not all the entites were loaded yet when this ran. Contrary to what the previous error said, I found that if we just wait then they will all be loaded eventually and this will return fully resolved.

Reply:
The one scenario where I've found where it comes up:

When you are referencing a deeply linked entry's asset and you update that asset in another tab, the live reload in the editor will run into this situation. With my code changes, it would error here, but now it will just throw a warning and eventually the linked asset will be loaded.